src/pages/MemberCollectionAdminPage/MemberCollectionAdminPage.tsx

```diff
@@ -115,25 +118,170 @@ const MemberCollectionAdminPage: React.FC = () => {
+  columns,
+}) => {
+  return (
+    <>
+      <div className="mr-2">{permissions.MEMBER_CREATE && <MemberCreationModal />}</div>
+      <div className="mr-2">
+        {exportImportVersionTag ? (
+          <MemberImportModal />
+        ) : (
+          <OldMemberImportModal /> // TODO: remove this after new export import completed
+        )}
+      </div>
+      {enabledModules.member_info_export ? (
+        exportImportVersionTag ? (
+          <MemberExportModal appId={appId} filter={fieldFilter} sortOrder={sortOrder} />
+        ) : (
+          <OldMemberExportModal // TODO: remove this after new export import completed
+            appId={appId}
+            visibleFields={visibleColumnIds}
+            columns={columns}
+            filter={fieldFilter}
+            sortOrder={sortOrder}
+          />
+        )
+      ) : null}
+    </>
+  )
+}
+
+export const MemberFieldFilter: React.FC<{
+  allColumns: ({
+    id: string
+    title: string
+  } | null)[]
+  visibleColumnIds: string[]
+  setVisibleColumnIds: (value: string[]) => void
+}> = ({ allColumns, visibleColumnIds, setVisibleColumnIds }) => {
+  const { formatMessage } = useIntl()
+  return (
+    <Popover
+      trigger="click"
+      placement="bottomLeft"
+      content={
+        <StyledOverlay>
+          <OverlayTitle>{formatMessage(memberMessages.label.fieldVisible)}</OverlayTitle>
+          <Checkbox.Group value={visibleColumnIds} onChange={value => setVisibleColumnIds(value as string[])}>
+            <FilterWrapper>
+              {allColumns.map(column =>
+                column ? (
+                  <Checkbox key={column.id} value={column.id}>
+                    {column.title}
+                  </Checkbox>
+                ) : null,
+              )}
+            </FilterWrapper>
+          </Checkbox.Group>
+        </StyledOverlay>
+      }
+    >
+      <StyledButton type="link" icon={<Icon component={() => <TableIcon />} />}>
+        {formatMessage(memberMessages.label.field)}
+      </StyledButton>
+    </Popover>
+  )
+}
+
+export const MemberCollectionBlock: React.VFC<
   Pick<AppProps, 'enabledModules' | 'settings'> & {
     currentUserRole: string
     appId: string
-    authToken: string
+    members: {
+      id: string
+      pictureUrl: string | null
+      name: string
+      email: string
+      role: 'general-member' | 'content-creator' | 'app-owner'
+      createdAt: Date
+      username: string
+      loginedAt: Date | null
+      managerId: string | null
+    }[]
+    loadingMembers: boolean
+    fieldFilter: FiledFilter
+    setFieldFilter: (filter: FiledFilter) => void
+    nextToken: string | null
+    limit: number
+    fetchMembers: (
+      filter: FiledFilter | undefined,
+      option: { limit?: number | undefined; nextToken?: string | null | undefined },
+    ) => Promise<ResponseMembers>
   }
-> = ({ currentUserRole, appId, enabledModules, settings, authToken }) => {
+> = ({
+  currentUserRole,
+  appId,
+  enabledModules,
```

Comment:
Is it unnecessary to be the props cause it's a global variable?

Reply:
Do you mean useContext?
It is indeed possible to get the value from useContext without using props, but this is the original way of writing it. I think it would be better to adjust it in refactor PR.

Reply:
ok, but I see that this part was modified, I think if you've modified here, you might as well handle this too.

Reply:
ok, modified.